src/framework/preferences/js/Builder.js

```diff
@@ -244,10 +276,27 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
     /*
      * A one-stop-shop function to build and instantiate a prefsEditor from a schema.
      */
-    fluid.prefs.create = function (container, options) {
-        options = options || {};
-        var builder = fluid.prefs.builder(options.build);
-        return fluid.invokeGlobalFunction(builder.options.assembledPrefsEditorGrade, [container, options.prefsEditor]);
+    fluid.defaults("fluid.prefs.create", {
+        gradeNames: ["fluid.viewComponent", "{that}.getGradeFromSchema"],
```

Comment:
It looks like this uses an unblessed technique of using a component's grade constructor as a free function - we will have better error checking in the future for this but this must never happen. Component creator functions must only ever be used by top-level code kicking off a standalone component tree construction - all other construction must occur declaratively.
It looks like we should return fluid.prefs.create into being a function, and then alter the action of fluid.prefs.builder so that it instead constructs some kind of linkage targetting fluid.uiOptions.prefsEditor rather than rewriting the contents of a statically named grade like fluid.prefs.constructed.prefsEditor - we should remove the latter from fluid.uiOptions.prefsEditor to avoid accidents. But fluid.prefs.create should be clear that it only advises the particular prefs editor that it creates itself, by running fluid.prefs.builder in a particular mode that doesn't cause the grade side-effects. The invocation of it in UIOptions.js should be provided some special option to indicate that it is intended to have the "spooky action at a distance" effect on fluid.uiOptions.prefsEditor

Reply:
Spoke with @amb26 about this in the [fluid-work channel today](http://irc-logs.fluidproject.org/%23fluid-work/%23fluid-work.2019-11-08.log).

Reply:
@amb26 what do you think of this idea.

- Have fluid.prefs.create (or rename to something more appropriate) be the top level component 
  - has a grade of fluid.primary.builder that merges the primary schemas
  - has a grade of something like fluid.primary.auxiliarybuilder that merges the auxiliary schemas
  - takes an array of preferences which are used to determine which primary and auxiliary schemas are added
  - has an invoker grade that generates and returns the synthesized grades to create the prefseditor, enhancer, and store sub components. Possibly an option to only create the store, prefseditor/store, enhancer/store, or all three.

This would keep a similar setup to the current PR where fluid.prefs.create is actually a component grade, the primary builder is likely not touched, the builder and auxbuilder are refactored (hopefully not too big a change), and the user of fluid.prefs.create wouldn't have to write anything that looked like a schema and could just request the preferences they want based on the preference type.